src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -194,6 +194,73 @@ protected override void Generate(MigrationOperation operation, IModel model, Mig
                     .Append(")");
             }
 
+            // PostgreSQL-XL - Distribute by (https://www.postgres-xl.org/documentation/sql-createtable.html)
+            if (operation[PostgresXlDistributeByAnnotationNames.DistributeBy] is string)
+            {
+                var distributeBy = new PostgresXlDistributeBy(operation);
+
+                var strategy = distributeBy.DistributionStrategy;
+                var distributeByColumn = distributeBy.DistributeByColumnName;
+                var distributeByFunction = distributeBy.DistributeByColumnFunction;
+                var distributionStyle = distributeBy.DistributionStyle;
+
+                if (distributionStyle == PostgresXlDistributionStyle.None)
+                {
+                    if (strategy == PostgresXlDistributeByStrategy.Replication
+                        || strategy == PostgresXlDistributeByStrategy.Roundrobin
+                        || (distributeByFunction != PostgresXlDistributeByColumnFunction.None
+                            && !string.IsNullOrWhiteSpace(distributeByColumn)))
+                    {
+                        if (strategy == PostgresXlDistributeByStrategy.Replication || strategy == PostgresXlDistributeByStrategy.Roundrobin)
+                        {
+                            builder.AppendLine()
+                                .Append("DISTRIBUTE BY ")
+                                .Append(strategy.ToString().ToUpperInvariant());
+                        }
+                        else if (distributeByFunction != PostgresXlDistributeByColumnFunction.None)
+                        {
+                            builder.AppendLine()
+                                .Append("DISTRIBUTE BY ")
+                                .Append(distributeByFunction.ToString().ToUpperInvariant())
```

Comment:
What happens if this is None?

Reply:
This can't be none as it is checked above